src/main/kotlin/no/nav/familie/oppdrag/simulering/SimulerBeregningRequestMapper.kt

```diff
@@ -0,0 +1,93 @@
+import no.nav.familie.oppdrag.iverksetting.*
+import no.nav.system.os.entiteter.typer.simpletypes.FradragTillegg
+import no.nav.system.os.entiteter.typer.simpletypes.KodeStatusLinje
+import no.nav.system.os.tjenester.simulerfpservice.simulerfpserviceservicetypes.Oppdrag
+import no.nav.system.os.tjenester.simulerfpservice.simulerfpserviceservicetypes.Oppdragslinje
+import org.springframework.stereotype.Component
+import java.time.format.DateTimeFormatter
+
+@Component
+class SimulerBeregningRequestMapper {
+    private val fpServiceGrensesnittFactory =
+            no.nav.system.os.tjenester.simulerfpservice.simulerfpservicegrensesnitt.ObjectFactory()
+    private val fpServiceTypesFactory = no.nav.system.os.tjenester.simulerfpservice.simulerfpserviceservicetypes.ObjectFactory()
+    private val oppdragSkjemaFactory = no.nav.system.os.entiteter.oppdragskjema.ObjectFactory()
+
+    private val tidspunktFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd-HH.mm.ss.SSSSSS")
+
+    fun tilSimulerBeregningRequest(utbetalingsoppdrag: Utbetalingsoppdrag): no.nav.system.os.tjenester.simulerfpservice.simulerfpservicegrensesnitt.SimulerBeregningRequest =
+            fpServiceGrensesnittFactory.createSimulerBeregningRequest()
+                    .apply { request = tilSimulerBeregningTypesRequest(utbetalingsoppdrag) }
+
+    private fun tilSimulerBeregningTypesRequest(utbetalingsoppdrag: Utbetalingsoppdrag): no.nav.system.os.tjenester.simulerfpservice.simulerfpserviceservicetypes.SimulerBeregningRequest =
+            fpServiceTypesFactory.createSimulerBeregningRequest().apply { oppdrag = tilOppdrag(utbetalingsoppdrag) }
+
+    private fun tilOppdrag(utbetalingsoppdrag: Utbetalingsoppdrag): Oppdrag {
+        oppdragSkjemaFactory.createAvstemmingsnokkel().apply {
+            avstemmingsNokkel = utbetalingsoppdrag.avstemmingTidspunkt.format(tidspunktFormatter)
+            kodeKomponent = utbetalingsoppdrag.fagSystem
+            tidspktReg = utbetalingsoppdrag.avstemmingTidspunkt.format(tidspunktFormatter)
+        }
+
+        val oppdragsEnhet = oppdragSkjemaFactory.createEnhet().apply {
+            enhet = OppdragSkjemaConstants.ENHET
+            typeEnhet = OppdragSkjemaConstants.ENHET_TYPE
+            datoEnhetFom = OppdragSkjemaConstants.ENHET_DATO_FOM.toString()
+        }
+
+        return fpServiceTypesFactory.createOppdrag().apply {
+            kodeEndring = EndringsKode.fromKode(utbetalingsoppdrag.kodeEndring.name).kode
+            kodeFagomraade = utbetalingsoppdrag.fagSystem
+            fagsystemId = utbetalingsoppdrag.saksnummer
+            utbetFrekvens = UtbetalingsfrekvensKode.MÅNEDLIG.kode
+            oppdragGjelderId = utbetalingsoppdrag.aktoer
+            datoOppdragGjelderFom = OppdragSkjemaConstants.OPPDRAG_GJELDER_DATO_FOM.toString()
+            saksbehId = utbetalingsoppdrag.saksbehandlerId
+            enhet.add(oppdragsEnhet)
+            avstemmingsnokkel.addAll(avstemmingsnokkel)
+            //ompostering
```

Comment:
Kan denne kommentaren fjernes? Vi har vel ikke støtte eller behov for omposteringer.